test/test_optim.py

```diff
@@ -839,18 +839,23 @@ def test_rmsprop(self):
 
     def test_asgd(self):
         for optimizer in [optim.ASGD, optim_mt.ASGD]:
+            if optimizer is optim_mt.ASGD:
```

Comment:
Why do you always skip the mt tests? That doesn't sound right?

Reply:
They have lower accuracy and I was told it was ok to do so as they are lower priority. I can likely get them to all pass if I bump up the `rtol` and `atol` for the mt tests.

Reply:
Well, the version without maximize did work before. We definitely don't want to remove these tests.

Reply:
The issue is it works but fails on rocm and windows. Is there an easy way to test in those environments as the mt tests pass in all environments I have access to?

Reply:
The original test before your change did work right? That part should not be removed.
For the newly added checks, you can have some restrictions. There are some variables in common_utils.py that can be used to know if you're running rocm or windows tests.

Reply:
I'll do that. Since, as you can see, it passes with the mt tests turned on now, but might fail for those tests later.